### ISSUE TITLE
checkpoint loading issue

### ISSUE THREAD

**Issue by Yangyx891121** [ctlllll/SGConv#6]
Greetings,

I ran into some trouble loading an SGConv network from checkpoint. Particularly, I encountered the following problem:

	size mismatch for kernel_norm: copying a param with shape torch.Size([2, 256, 1]) from checkpoint, the shape in current model is torch.Size([256, 1]).

A few things I have noticed:
* The above error can be replicated by running the following code in ipynb.
* In the code below, the error goes away if I forward the layer before loading.

Any pointers on what caused this error and how it can be solved are greatly welcome. Thanks!

------

import torch
from gconv_standalone import GConv

layer = GConv(
    d_model=256,
    d_state=64,
    l_max=1_000_000,
    bidirectional=True,
    kernel_dim=32,
    n_scales=None,
    decay_min=2,
    decay_max=2,
)

x = torch.randn(1, 256, 1000)
x = x.cuda()
layer.cuda()
y, k = layer(x, return_kernel=True)

path = './dummy_ckpt'
torch.save({
    'state_dict': layer.state_dict()
}, path)

shell = GConv(
    d_model=256,
    d_state=64,
    l_max=1_000_000,
    bidirectional=True,
    kernel_dim=32,
    n_scales=None,
    decay_min=2,
    decay_max=2,
).cuda()

ckpt = torch.load(path)
shell.load_state_dict(ckpt['state_dict'])

**Comment by ctlllll** [comment]
Hi Yingxiang, thanks for your interest, and thank you for reporting this bug! It's an issue related to the kernel_norm buffer which we used for normalization. Please pull the new commits and see whether it works now.

**Comment by ghost** [comment]
Thank you very much for the prompt response! It worked.
